src/components/SourcesTree.js

```diff
@@ -97,6 +100,18 @@ let SourcesTree = React.createClass({
       parentMap: createParentMap(sourceTree) });
   },
 
+  componentDidUpdate(prevProps, prevState) {
+    // Restore tabs from localStorage
+    if (typeof localStorage !== "undefined" &&
+    this.props.sources == prevProps.sources &&
+    !this.props.hasRestoredTabs) {
+      const tabs = JSON.parse(localStorage.getItem("tabs"));
+      if (tabs) {
+        tabs.forEach(url => this.props.selectSourceByURL(url));
+      }
+    }
+  },
```

Comment:
couple things:

1. we should do the JSON.parse in the reducer
2. we *dont* need to select all of the sources. in fact this is very slow. We should be able to just show the tabs if it's in the appropriate state